src/libtools/edgecontainer.cc

```diff
@@ -37,11 +37,29 @@ EdgeContainer::EdgeContainer(vector<Edge> eds) {
   }
 }
 
+int EdgeContainer::getMaxDegree(void){
+  int max = 0;
+  for(auto const& it : adj_list_) {
+    if(lexical_cast<int>(it.second.size())>max) max = it.second.size();
```

Comment:
why? what the `lexical_cast` is doing here?

Reply:
I guess this explains why I had the lexical cast, size() returns type `size_t` whereas max is of type int:

[  2%] Building CXX object tools/src/libtools/CMakeFiles/votca_tools.dir/graph_bf_visitor.cc.o
/home/votca/votca/tools/src/libtools/edgecontainer.cc: In member function 'int votca::tools::EdgeContainer::getMaxDegree()':
/home/votca/votca/tools/src/libtools/edgecontainer.cc:43:24: error: comparison of integer expressions of different signedness: 'std::set<int>::size_type' {aka 'long unsigned int'} and 'int' [-Werror=sign-compare]
     if(it.second.size()>max) max = it.second.size();
        ~~~~~~~~~~~~~~~~^~~~
cc1plus: all warnings being treated as errors
make[2]: *** [tools/src/libtools/CMakeFiles/votca_tools.dir/build.make:167: tools/src/libtools/CMakeFiles/votca_tools.dir/edgecontainer.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:244: tools/src/libtools/CMakeFiles/votca_tools.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
The command '/bin/sh -c make -j2' returned a non-zero code: 2